### PR TITLE
Factor GCC recipes and fix configuration

### PR DESCRIPTION
**Pull request by alarcher**

### CODE REVIEW

**Comment by alarcher** [comment]
@richlowe If you have any comment.

File: make-rules/gcc-component.mk [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+COMPONENT_SUMMARY= GNU Compiler Collection
+COMPONENT_CLASSIFICATION= Development/C
+COMPONENT_PROJECT_URL = http://gcc.gnu.org/
+COMPONENT_SRC ?= $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE ?= $(COMPONENT_SRC).tar.xz
+COMPONENT_ARCHIVE_URL ?= \
+  http://ftp.gnu.org/gnu/gcc/gcc-$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+
+PATCH_EACH_ARCHIVE=1
+PATCHDIR_PATCHES = $(shell find $(PATCH_DIR) $(PARFAIT_PATCH_DIR) -type f -name '$(PATCH_PATTERN)' \
```
Comment (richlowe) [inline make-rules/gcc-component.mk]: Might as well remove any support for Parfait

File: make-rules/gcc-component.mk [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2019 Aurelien Larcher
+#
+
+GCC_COMPONENT_VERSION_MAJOR = $(shell echo $(COMPONENT_VERSION) | $(GSED) -e 's/\([0-9]\+\)\.[0-9]\+.*/\1/')
```
Comment (richlowe) [inline make-rules/gcc-component.mk]: `awk -F. '{print $1}'`?

**Comment by alarcher** [comment]
I have rebuilt all of them so I am going to merge soon if it is OK with you.

**Comment by richlowe** [comment]
_ME_ me?!  It's fine with me, but maybe someone more actively OI-related would be good approving things too? :)

**Review by Mno-hime — APPROVED** [review]

**Comment by alarcher** [comment]
@richlowe Yes YOU you :) and @Mno-hime also approved, all good :) now I hope the build server will not curse me again with a failure while everything worked out perfectly in my build zone...

**Pull request merged**